components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -7057,6 +7059,50 @@ public static void deletePolicyPartialMappings(Integer policyGroupId, Connection
 		}
 	}
 
+    /**
+     * Remove XACML Policies from Entitlement Service
+     *
+     * @param policyGroupId
+     * @param conn
+     * @throws SQLException
+     */
+    public static void deleteXACMLPoliciesFromEntitlementService(Integer policyGroupId, Connection conn)
+            throws SQLException {
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        String query = " SELECT POLICY_ID FROM APM_POLICY_GRP_PARTIAL_MAPPING WHERE POLICY_GRP_ID=? ";
+        String policyId = "";
+
+        //Define Entitlement Service
+        AppManagerConfiguration config = ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration();
+        EntitlementService entitlementService = EntitlementServiceFactory.getEntitlementService(config);
+
+        try {
+            ps = conn.prepareStatement(query);
+            ps.setInt(1, policyGroupId);
+            rs = ps.executeQuery();
+            while (rs.next()) {
+                policyId = rs.getString("POLICY_ID");
+                //If policy id not null, remove the Entitlement policy with reference to policy id
+                if (policyId != null) {
+                    entitlementService.removePolicy(policyId);
+                }
+            }
+        } catch (SQLException e) {
```

Comment:
This is a private method. SQL exception is thrown to a DAO layer level function to handle transactions properly.